[Weather] Add feels-like temperature display

RKBoss6 opened the pull request.

stweedo commented:
I just checked your version of the shortcut and it was missing the “mostly cloudy” filter that shows multiple clouds on the bangle. I added your changes as well and this version contains everything. https://www.icloud.com/shortcuts/71f49664b5a44b6cb80703b73af2d8f0

RKBoss6 commented:
Alright @stweedo thanks!

Rengyr commented:
There might be support for "feels like temp" for android users later (and if users will use the new GadgetBridge extended weather data, which is optional setting). So best would be to limit not it based on the platform, but just check if the field exists in the weather data we get. Either we will have value or the field will be undefined, in which case we could just hide it?

bobrippling commented on apps/weather/app.js:
```diff
@@ -9,59 +9,67 @@ Bangle.loadWidgets();
-        {type: "txt", font: "6x8", pad: 2, valign: -1, id: "windUnit", label: "km/h"},
-      ]},
-      {type: "custom", fillx: 1, height: 15, id: "uvDisplay",
+        render: l => weather.drawIcon(l, l.x+l.w/2, l.y+l.h/2.1, l.w/2.1-10)},
```
I guess this is the icon that's not clearing? The `custom` layout will [clear it for you if you set `bgCol`](https://www.espruino.com/Bangle.js+Layout#layout-positioning), then so long as you draw in the bounds I'd expect you to be alright

RKBoss6 replied:
Hey! I tried that, but it didn't look like it was being drawn in the bounds. How would you change the bounds for the icon? I'm asking because if we fit the icon in the bounds, there is a lot of wasted space that can be minimized to get a larger icon.

bobrippling replied:
Interesting - I'm afraid i'm not great on icons, @gfwilliams do you know?

gfwilliams replied:
I think they're treated like any other element in layout, it's just that the icon is drawn in the middle. So if you set other elements in the line to fillx:1 then they will fill any excess space

RKBoss6 commented:
Alright, this should be good to go now. I fixed the icon clearing issue, by making it slightly smaller to fit inside the bounds, and changed the expiry text to be ambiguous to both iOS and Android.
